eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/ConcurrentHashMapUnsafe.java

```diff
@@ -909,11 +911,12 @@ public V put(K key, V value)
         if (o == null)
         {
             Entry<K, V> newEntry = new Entry<>(key, value, null);
+            this.addToSize(1);
             if (ConcurrentHashMapUnsafe.casArrayAt(currentArray, index, null, newEntry))
             {
-                this.addToSize(1);
                 return null;
             }
+            this.addToSize(-1);
```

Comment:
This means if you insert one key, then insert a second colliding key, you could observe for a moment a size of 2?

Reply:
Correct. That was preferable to seeing an empty map (size 0) in the same thread that's doing a simple put.